src/providers/lingyi/chatComplete.ts

```diff
@@ -0,0 +1,137 @@
+import { LINGYI } from "../../globals";
+
+import {
+  ChatCompletionResponse,
+  ErrorResponse,
+  ProviderConfig,
+} from "../types";
+
+export const LingYiChatCompleteConfig: ProviderConfig = {
+  model: {
+    param: "model",
+    required: true,
+    default: "yi-34b-chat-0205",
+  },
+  messages: {
+    param: "messages",
+    default: "",
+  },
+  max_tokens: {
+    param: "max_tokens",
+    default: 100,
+    min: 0,
+  },
+  temperature: {
+    param: "temperature",
+    default: 1,
+    min: 0,
+    max: 2,
+  },
+  top_p: {
+    param: "top_p",
+    default: 1,
+    min: 0,
+    max: 1,
+  },
+  stream: {
+    param: "stream",
+    default: false,
+  },
+};
+
+export interface LingYiChatCompleteResponse extends ChatCompletionResponse {}
+
+export interface LingYiErrorResponse extends ErrorResponse {}
+
+export interface LingYiStreamChunk {
+  id: string;
+  object: string;
+  created: number;
+  model: "yi-34b-chat-0205" | "yi-34b-chat-200k" | "yi-vl-plus";
+  choices: {
+    delta: {
+      content?: string;
+    };
+    index: number;
+    finish_reason: string | null;
+  }[];
+}
+
+export const LingYiChatCompleteResponseTransform: (
+  response: LingYiChatCompleteResponse | LingYiErrorResponse,
+  responseStatus: number,
+) => ChatCompletionResponse | ErrorResponse = (response, responseStatus) => {
+  if ("error" in response && responseStatus !== 200) {
+    return {
```

Comment:
You can use `generateErrorResponse` to generate the provider error response. Applicable for all the provider integrated in this PR.
```
return generateErrorResponse(
  {
    message: response.error.message,
    type: response.error.type,
    param: null,
    code: response.error.code?.toString() || null,
  },
  LINGYI
);
```